src/main/java/com/example/repository/BookSpecificationBuilder.java

```diff
@@ -0,0 +1,35 @@
+package com.example.repository;
+
+import com.example.dto.BookSearchParameters;
+import com.example.model.Book;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Component;
+
+@RequiredArgsConstructor
+@Component
+public class BookSpecificationBuilder implements SpecificationBuilder<Book> {
+    private static final String TITLE = "title";
+    private static final String AUTHOR = "author";
+    private static final String ISBN = "isbn";
+
```

Comment:
better make it public and reuse it in Providers (and have one place for future changes)